packages/catalog-process/test/addEServiceTemplateInstanceInterface.test.ts

```diff
@@ -0,0 +1,663 @@
+  const filePath = `./resources/${fileName}`;
+
+  const fileContent = await fs.readFile(`${dirname}/${filePath}`);
+  return fileContent.toString();
+};
+
+const initEserviceTemplateInstance = async (
+  eserviceDescriptorState: DescriptorState,
+  technology?: Technology,
+  interfaceDoc?: {
+    doc: Document;
+    content: string;
+  }
+): Promise<{
+  eservice: EService;
+  descriptor: Descriptor;
+  template: EServiceTemplate;
+}> => {
+  const interfacePath = interfaceDoc
+    ? await fileManager.storeBytes(
+        {
+          bucket: config.s3Bucket,
+          path: interfaceDoc.doc.path,
+          resourceId: interfaceDoc.doc.id,
+          name: interfaceDoc.doc.name,
+          content: Buffer.from(interfaceDoc.content),
+        },
+        genericLogger
+      )
+    : undefined;
+
+  const mockEserviceTemplateVersion: EServiceTemplateVersion = {
+    ...getMockEServiceTemplateVersion(
+      generateId<EServiceTemplateVersionId>(),
+      eserviceTemplateVersionState.published
+    ),
+    interface:
+      interfaceDoc?.doc && interfacePath
+        ? { ...interfaceDoc.doc, path: interfacePath }
+        : undefined,
+  };
+
+  const mockEServiceTemplate = {
+    ...getMockEServiceTemplate(
+      generateId<EServiceTemplateId>(),
+      generateId<TenantId>(),
+      [mockEserviceTemplateVersion]
+    ),
+    interface: interfaceDoc,
+    technology: technology || "Rest",
+  };
+
+  const mockDescriptor = {
+    ...getMockDescriptor(eserviceDescriptorState),
+    templateVersionRef: {
+      id: mockEServiceTemplate.versions[0].id,
+    },
+  };
+
+  const mockEService = {
+    ...getMockEService(),
+    descriptors: [mockDescriptor],
+    technology: technology || "Rest",
+    templateRef: {
+      id: mockEServiceTemplate.id,
+    },
+  };
+
+  await addOneEService(mockEService);
+  await addOneEServiceTemplate(mockEServiceTemplate);
+
+  return {
+    eservice: mockEService,
+    descriptor: mockDescriptor,
+    template: mockEServiceTemplate,
+  };
+};
+
+describe("addEServiceTemplateInstanceInterface", () => {
+  beforeAll(() => {
+    vi.useFakeTimers();
+    vi.setSystemTime(new Date());
+  });
+
+  afterAll(() => {
+    vi.useRealTimers();
+  });
+
+  describe("Invalid data input (Rest/Soap)", () => {
+    it("should throw an eServiceNotFound if template doesn't contains serverUrls", async () => {
+      const eserviceId = generateId<EServiceId>();
+      await expect(
+        catalogService.addEServiceTemplateInstanceInterface(
+          eserviceId,
+          generateId(),
+          {
+            contactName: "Jhon Doe",
+            contactUrl: "https://fun.tester.johnny.info",
+            contactEmail: "johnnyd@funnytester.com",
+            termsAndConditionsUrl: "https://fun.tester.johnny.terms.com",
+            serverUrls: ["https://fun.tester.server.com"],
+          },
+          getMockContext({ authData: getMockAuthData() })
+        )
+      ).rejects.toThrow(eServiceNotFound(eserviceId));
+    });
```

Comment:
🤔  I still don't understand this test: it says "if template doesn't contains serverUrls" but it seems it's not retrieving any template, and also if the issue is on serverUrls in the template, it should not throw `eServiceNotFound`. If we want to test the case where the e-service cannot be found this should probably become something like `"should throw an eServiceNotFound if the e-service does not exist"`

Reply:
In this commit [ec51f08](https://github.com/pagopa/interop-be-monorepo/pull/1697/commits/ec51f087f009097ea911adfdea0a50d8625770b7) this description was properly changed to "should throw an eServiceNotFound if the e-service does not exist"